lib/IDE/CodeCompletion.cpp

```diff
@@ -1753,35 +1753,29 @@ class CompletionLookup final : public swift::VisibleDeclConsumer {
   }
 
   void addImportModuleNames() {
-    // FIXME: Add user-defined swift modules
-    SmallVector<StringRef, 20> ModuleNames;
+    SmallVector<Identifier, 0> ModuleNames;
+    Ctx.getVisibleTopLevelModuleNames(ModuleNames);
 
-    // Collect clang module names.
-    {
-      SmallVector<clang::Module*, 20> ClangModules;
-      Ctx.getVisibleTopLevelClangModules(ClangModules);
-      for (auto *M : ClangModules) {
-        if (!M->isAvailable())
-          continue;
-        if (M->getTopLevelModuleName().startswith("_"))
-          continue;
-        if (M->getTopLevelModuleName() == Ctx.SwiftShimsModuleName.str())
-          continue;
-
-        ModuleNames.push_back(M->getTopLevelModuleName());
-      }
-    }
+    llvm::StringSet<> ImportedModules;
+    collectImportedModules(ImportedModules);
 
     std::sort(ModuleNames.begin(), ModuleNames.end(),
-              [](StringRef LHS, StringRef RHS) {
-                return LHS.compare_lower(RHS) < 0;
+              [](Identifier LHS, Identifier RHS) {
+                return LHS.str().compare_lower(RHS.str()) < 0;
               });
+    ModuleNames.erase(std::unique(ModuleNames.begin(), ModuleNames.end()),
+                      ModuleNames.end());
 
-    llvm::StringSet<> ImportedModules;
-    collectImportedModules(ImportedModules);
-
+    auto mainModuleName = CurrDeclContext->getParentModule()->getName();
     for (auto ModuleName : ModuleNames) {
-      auto MD = ModuleDecl::create(Ctx.getIdentifier(ModuleName), Ctx);
+      if (ModuleName == mainModuleName)
+        continue;
+      if (ModuleName.str().startswith("_"))
+        continue;
+      if (ModuleName == Ctx.SwiftShimsModuleName)
+        continue;
```

Comment:
Are there other module names we should hide? e.g. `SwiftOnoneSupport`

Reply:
+1 to skipping OnoneSupport (looks like the name is in Strings.h).  Other modules we should skip: `__C`, `__C_Synthesized`, `Builtin`.  I'm guessing they are already hidden because they're not on the file system, but please add `CHECK-NOT`s for them.

Reply:
`_InternalSwiftSyntaxParser` is another one we don't want to show. Should already be covered, but again, worth a test to make sure.